src/metabase/search/appdb/index.clj

```diff
@@ -265,6 +265,13 @@
              (partial merge-with +)
              document-reducible))
 
+(defmethod search.engine/delete! :search.engine/appdb [_engine model ids]
+  (when-let [index-table (active-table)]
+    (t2/query {:delete-from index-table
+               :where       [:and
+                             [:= :model model]
+                             [:in :model_id ids]]})))
```

Comment:
Ok now you're making me dig up the old code I deleted, cos it was still better than this 😉

Reply:
Also we should delete from pending as well, cos we wouldn't want it to come back from the data 🧟

Reply:
Damn I can't find it, but essentially determines the `search-model` (s) by filtering the specifications for matches on `:model`.

Reply:
Well this should be done higher up, before it gets to the index, right? So that engines do not repeat the logic.

I thought about doing this, but... Should it look at "(model-hooks)"? And how do you know what your incoming "ids" are about?